src/compiler/utilities.ts

```diff
@@ -6258,6 +6258,34 @@ namespace ts {
         return isSourceFile(node) || isModuleBlock(node) || isBlock(node) && isFunctionLike(node.parent);
     }
 
+    /* @internal */
+    export function isNewThisScope(node: Node): boolean {
+        switch (node.kind) {
+            case SyntaxKind.ArrowFunction:
+                switch (node.parent.kind) {
+                    case SyntaxKind.PropertyDeclaration:
+                        return true;
+                    default:
+                        return false;
+                }
+            case SyntaxKind.FunctionDeclaration:
+            case SyntaxKind.FunctionExpression:

```

Comment:
Now that you removed classes, you need to add `ConstructorDeclaration` and `PropertyDeclaration`.
The latter one makes the `case SyntaxKind.ArrowFunction` redundant.